src/server/project.ts

```diff
@@ -48,6 +48,11 @@ namespace ts.server {
         return counts.js > 0 && counts.ts === 0 && counts.tsx === 0;
     }
 
+    export function hasBothJsAndTsFiles(project: Project) {
+        const counts = countEachFileTypes(project.getRootScriptInfos());

```

Comment:
So does this only look at root files? What app when someone just compiles an entry point (e.g. `app.ts`) and the rest are resolved by imports (or references)? Basing the error on whether an output file is going to overwrite an input file (which we already have an error for) seems simpler and catches more scenarios that are misconfigured.